src/verifier/types.ts

```diff
@@ -0,0 +1,33 @@
+import { LogLevel } from '../service';
+
+export interface ConsumerVersionSelector {
+  pacticipant?: string;
+  tag?: string;
+  version?: string;
+  latest?: boolean;
+  all?: boolean;
+}
+
+export interface VerifierOptions {
+  providerBaseUrl: string;
+  provider?: string;
+  pactUrls?: string[];
+  pactBrokerUrl?: string;
+  pactBrokerUsername?: string;
+  pactBrokerPassword?: string;
+  pactBrokerToken?: string;
+  consumerVersionTags?: string | string[];
+  providerVersionTags?: string | string[];
+  providerStatesSetupUrl?: string;
+  publishVerificationResult?: boolean;
+  providerVersion?: string;
+  enablePending?: boolean;
+  // Todo in FFI
+  includeWipPactsSince?: string;
+  consumerVersionSelectors?: ConsumerVersionSelector[];
+  logLevel?: LogLevel;
```

Comment:
So a more general point is that we have access to these logging facilities now: https://github.com/pact-foundation/pact-reference/blob/master/rust/pact_mock_server_ffi/src/lib.rs#L88-L97.

There is still an outstanding PR to enable logging into it from a remote interface (this was the [WIP PR](https://github.com/pact-foundation/pact-reference/pull/105) to demonstrate the concept)

This property (or something like it) should be supported. We have a few choices, I think we were trending towards one of the following:

1. Log everything through the rust interface and print to std out
2. Log everything through the rust interface and log to file
3. Log everything through the rust interface (via `log_to_buffer`, `log`) but fetch the buffer at the end of a run per mock server via `fetch_log_buffer` and control the output via pact-core

I think 3 would be ideal.

Reply:
I actually prefer options 1 and/or 2 - if something explodes mid run, we've got the best chance to have the logs. And, if the pattern is to defer logging to the core, then we'll get more consistency across pact projects in general.

Reply:
Whichever option we choose, I think they are all clear arguments for a single ffi lib.

Reply:
I agree on the single FFI lib. That will take a bit of work, but not massive amounts. 

In the meantime, the log functions are all exposed in the mock server lib, which will apparently (untested by me) work as if it were a single application anyway. 

Good point on the crash and consistency. FWIW it's what I'm doing with Go. I have the output in the `go test` looking pretty nice now, so a separate log file or just enabling a more verbose log level is probably going to be just fine.

The log function I originally proposed lets you log the context across the boundary e.g. something like `pact-js-core:<package>:<subpackage>` could prefix all log entries, so you could quickly identify where to debug next.

Reply:
I'm actually wondering if we remove this property altogether, and just have a global logger interface that lets us set the level. Then it can be independent of the verifier and the mock server - what do you think?

Reply:
I think most people are likely to use the constructors or factories as entry points into pact - and it's convenient for those to have a `logLevel` property. Even though we expose multiple tools, typically people are only using one of them in a given context, I think. But, I do think that we can do better with config, in general. The config proposal is related, here: https://github.com/pact-foundation/pact-js/issues/602